### DISCUSSION TITLE
How to enable Front Channel Logout without post_logout_redirection_uri parameter

### DISCUSSION THREAD

**Discussion by ssamsundar** [keycloak/keycloak#14193] [Q&A]
I'm using Keycloak docker version 18.0.2.

When I try to logout from the application, it must logout from external idp as well. I tried using front-channel logout option with id_token_hint, but it also uses post_logout_redirection_uri.
I don't want to redirect back to keycloak, rather it must stay in external idp logout screen.

How to disable the post_logout_redirection_uri parameter from logout flow?

**Comment by lexcao** [comment]
What identity provider are you configured with?
If it is OIDC, you could keep `Logout URL` empty to skip logout from IDP.
If it is SAML, you could keep `Single Logout Service URL` empty to skip the logout.

**Reply by ssamsundar** [reply]
I'm using OIDC with front-channel logout and I want to logout from both keycloak & external idp as well.

When I try to do logout from keycloak

https://myapp.mycompany.com/keycloak/realms/myapp/protocol/openid-connect/logout?id_token_hint=<id_token>

It automatically redirects to external idp logout

https://company-oidc.mycompany.com/openam/oauth2/realms/root/realms/AccessManagement/connect/endSession?state=<stateid>&id_token_hint=<idp_id_token>**&post_logout_redirect_uri=https%3A%2F%2Fmyapp.mycompany.com%2Fkeycloak%2Frealms%2Fmyapp%2Fbroker%2Foidc%2Fendpoint%2Flogout_response**

Now, I want to remove the **post_logout_redirect_uri** parameter from the external idp.
https://company-oidc.mycompany.com/openam/oauth2/realms/root/realms/AccessManagement/connect/endSession?state=<stateid>&id_token_hint=<idp_id_token>

@lexcao whether is it possible to make post logout redirect uri as optional and turn it off?

**Reply by lexcao** [reply]
From my understanding,
currently, there is no button to control the `post_logout_redirect_uri` parameter.
I think it is a good idea to support such a feature.
Need Keycloak team to check this.

**Comment by ssamsundar** [comment]
Hi,

Sorry for the confusion here.

I'm using OIDC. i want to logout from external idp but i don't want to send
post logout redirection uri parameter during the logout.
